ui/src/DashboardTable.js

```diff
@@ -0,0 +1,416 @@
+    fetch(`${FETCH_PATH}/api/models_paginated/${page * rowsPerPage}/${page * rowsPerPage + rowsPerPage}`)
+      .then((response) => response.json())
+      .then((json) => {
+        setRows(json["models"]);
+        setCount(json["count"]);
+        json["models"].map((m) => {
+          if (m.progress < 100) {
+            setProgress(progress + m.progress + 3);
+          }
+        })
+      });
+  }, [page, rowsPerPage, progress]);
+
+
+  const handleSelectAllClick = (event) => {
+    if (event.target.checked) {
+      const newSelected = rows.map((n) => n.id);
+      setSelected(newSelected);
+      return;
+    }
+    setSelected([]);
+  };
+
+  const handleClick = (event, filename) => {
+    const selectedIndex = selected.indexOf(filename);
+    let newSelected = [];
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, filename);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1),
+      );
+    }
+
+    setSelected(newSelected);
+  };
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeDense = (event) => {
+    setDense(event.target.checked);
+  };
+
+  const isSelected = (name) => selected.indexOf(name) !== -1;
+
+  // Avoid a layout jump when reaching the last page with empty rows.
+  const emptyRows =
+    page > 0 ? Math.max(0, (1 + page) * rowsPerPage - rows.length) : 0;
+
+  useEffect(() => {
+    fetch(`${FETCH_PATH}/api/delete/${deleted}`, {
+      method: 'POST'
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        setSelected([])
+      });
+  }, [deleted]);
+
+  function onDelete() {
+    setDeleted(selected.join('.'))
+  }
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      <Paper sx={{ width: '100%', mb: 2 }}>
+        <EnhancedTableToolbar numSelected={selected.length} onDelete={onDelete} />
+        <TableContainer>
+          <Table
+            sx={{ minWidth: 750 }}
+            aria-labelledby="tableTitle"
+            size={dense ? 'small' : 'medium'}
+          >
+            <EnhancedTableHead
+              numSelected={selected.length}
+              onSelectAllClick={handleSelectAllClick}
+              rowCount={rows.length}
+            />
+            <TableBody>
+              {/* if you don't need to support IE11, you can replace the `stableSort` call with:
+                 rows.sort(getComparator(order, orderBy)).slice() */}
+              {rows.map((row, index) => {
+                  const isItemSelected = isSelected(row.id);
+                  const labelId = `enhanced-table-checkbox-${index}`;
+                  if (row.progress == 100) {
```

Comment:
Let's keep it like this for now, but i'd rather have the if statement on some cells than the entire row, because there is quite a bit of duplication now between the branches.

Reply:
Fixed in https://github.com/AECgeeks/ifc-pipeline-validation/pull/51/commits/25a4f3721a8caeb4b25d0cabd7e755fc06b15464